external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisInputDStream.scala

```diff
@@ -249,6 +252,17 @@ object KinesisInputDStream {
     }
 
     /**
+      * Sets the [[SparkAWSCredentials]] to use for authenticating to the AWS CloudWatch
+      * endpoint. Will use the same credentials used for AWS Kinesis if no custom value is set.
+      *
+      * @param conf: Map[String, String] to use for CloudWatch authentication
+      */
+    def kinesisConf(conf: Map[String, String]): Builder = {
```

Comment:
Do you think it would be better to pass values to builder rather than a map of configs. I thought map of configs can be easily extended when we need to support new configurations without code changes?
What is your thought on values+builder per config vs one map for all configs ?